controller/linodecluster_controller.go

```diff
@@ -128,14 +123,15 @@ func (r *LinodeClusterReconciler) reconcile(
 		}
 	}()
 
+	if err := clusterScope.AddFinalizer(ctx); err != nil {
+		return res, err
+	}
+
 	// Handle deleted clusters
 	if !clusterScope.LinodeCluster.DeletionTimestamp.IsZero() {
 		return res, r.reconcileDelete(ctx, logger, clusterScope)
 	}
 
-	if err := clusterScope.AddFinalizer(ctx); err != nil {
```

Comment:
is there a reason we want to move the finalizer down bellow the delete?

Reply:
Oh, I accidentally moved it before the delete step. I'll move it below it.